lib/marc_cleanup/variable_fields.rb

```diff
@@ -1,5 +1,122 @@
 module MarcCleanup
 
+### Remove non-numberical strings and append a new 020$q with the string
+def new_020_q(record)
+  record.fields('020').each do |f020|
+    f020.subfields.each do |subfield|
+      next unless subfield.code == 'a'
+      isbn_parts = /^\s*([\d]+)\s*(\(.*?\))\s*$/.match(subfield.value)
+      next if isbn_parts.nil?
+      subfield.value = isbn_parts[1]
+      f020.append(MARC::Subfield.new('q', isbn_parts[2])) 
+    end  
+  end
+  record
+end
+
+### Convert ISBN-10 to ISBN-13
+def isbn10_to_13(isbn)
+  stem = isbn[0..8]
+  return nil if stem =~ /\D/
+
+  existing_check = isbn[9]
+  return nil if existing_check && existing_check != checkdigit_10(stem)
+
+  main = ISBN13PREFIX + stem
+  checkdigit = checkdigit_13(main)
+  main + checkdigit
+end
+
+### Calculate check digit for ISBN-10
+def checkdigit_10(stem)
+  int_index = 0
+  int_sum = 0
+  stem.each_char do |digit|
+    int_sum += digit.to_i * (10 - int_index)
+    int_index += 1
+  end
+  mod = (11 - (int_sum % 11)) % 11
+  mod == 10 ? 'X' : mod.to_s
+end
+
+### Calculate check digit for ISBN-13
+def checkdigit_13(stem)
+  int_index = 0
+  int_sum = 0
+  stem.each_char do |digit|
+    int_sum += int_index.even? ? digit.to_i : digit.to_i * 3
+    int_index += 1
+  end
+  ((10 - (int_sum % 10)) % 10).to_s
+end
+
+### Normalize ISBN-13
+def isbn13_normalize(raw_isbn)
+  int_sum = 0
+  stem = raw_isbn[0..11]
+  return nil if stem =~ /\D/
+
+  int_index = 0
+  stem.each_char do |digit|
+    int_sum += int_index.even? ? digit.to_i : digit.to_i * 3
+    int_index += 1
+  end
+  checkdigit = checkdigit_13(stem)
+  return nil if raw_isbn[12] && raw_isbn[12] != checkdigit
+
+  stem + checkdigit
```

Comment:
According to coveralls, line 67 is not currently tested. That means the tests aren't reaching this line in the code. I'll leave a summary of the cases that need to be tested in a summary comment for the review.